### PR TITLE
Added overload of Refresh for properly changing background/foreground colours for wxSpinCtrl (MSW)

### PR DESCRIPTION
**Pull request by Kvaz1r**
Patch for [ticket 12382 - Can't change wxSpinCtrl background or foreground color](https://trac.wxwidgets.org/ticket/12382)

### CODE REVIEW

**Comment by vadz** [comment]
Could you please explain how does this work? Looking at the code changes, I just don't see it... how exactly does it change the colour of the buddy control?

Also, we don't call `Refresh()` from base `SetXXXColour()` implementation (the idea being that it's common to change both colours at once), so I don't think we should call it from here neither.

**Comment by Kvaz1r** [comment]
> Could you please explain how does this work? Looking at the code changes, I just don't see it... how exactly does it change the colour of the buddy control?

As far I understand(based on behaviour sample on Win7), buddy update colour with spin control, but it's not visible since it out of main control area so didn't refresh. But it will refresh with some event (change text, spin event, mouse hower,...).  So I added force refresh for buddy control when spin change colour.

**Comment by vadz** [comment]
Normally `SetXXXColour()` + `Refresh()` is supposed to work. So perhaps it's `Refresh()` that should be overridden in `wxSpinCtrl` to also refresh the buddy control?

**Comment by Kvaz1r** [comment]
Yes, it's even better. I'll update the PR.

File: src/msw/spinctrl.cpp [hunk 1]
```diff
@@ -404,6 +404,12 @@ wxSpinCtrl::~wxSpinCtrl()
     gs_spinForTextCtrl.erase(GetBuddyHwnd());
 }
 
+void wxSpinCtrl::Refresh(bool eraseBackground, const wxRect *rect)
+{
+    wxControl::Refresh(eraseBackground, rect); 
+    ::RedrawWindow(GetBuddyHwnd(), 0, 0, RDW_INVALIDATE | RDW_UPDATENOW);
```
Comment (vadz) [inline src/msw/spinctrl.cpp]: Does it still work without `RDM_UPDATENOW`? We don't use it in the base class `Refresh()`, so it seems strange to use it just for the buddy control and I'd prefer to remove it if things still work without it.

OTOH we could add `RDW_ERASE` if `eraseBackground == true`, again, for consistency with the default implementation.

Comment (Kvaz1r) [reply src/msw/spinctrl.cpp]: Yes, it works. Fixed.

**Comment by vadz** [comment]
Thanks!

**Pull request closed without merging**